src/Xamarin.Android.Tools.Aidl/CSharpCodeGenerator.cs

```diff
@@ -613,7 +630,15 @@ string GetWriteStatements (TypeName type, string parcel, string arg, string parc
 					return String.Format ("{1}.WriteStrongBinder (((({0} != null)) ? ({0}.AsBinder ()) : (null)));", arg, parcel);
 			}
 		}
-		
+
+		string GetWriteOutStatements (TypeName type, string parcel, string arg)
+		{
+			if (type.ArrayDimension > 0) {
+				return "if (" + arg + " == null) { " + parcel + ".WriteInt(-1); } else { " + parcel + ".WriteInt(" + arg + ".Length); }";
```

Comment:
Given that `.WriteInt(-1)` doesn't show up anywhere in our current expected test output, we're going to need a new unit test which ensures this code path is hit.

Reply:
@mfranke-moba : please add a unit test which causes this line to be triggered.

Reply:
This is already covered by `Test11`. The `WriteInt(-1)` will be written for out array parameters like `out byte[]`. In this case no byte array is written with `WriteXyzArray` like for `in byte[]` or `inout byte[]` as no data has to be transmitted, just the size.

Reply:
Ah, I had missed that. :-(